pkg/virtctl/config_ssh/config_ssh.go

```diff
@@ -0,0 +1,221 @@
+			if err != nil {
+				return fmt.Errorf("failed to fetch %s: %v", gvr.Resource, err)
+			}
+
+			hosts, err := generateHostEntries(programName, currentContext, objects.Items)
+			if err != nil {
+				return err
+			}
+			cfg.Hosts = append(cfg.Hosts, hosts...)
+		}
+	}
+
+	if dryRun {
+		fmt.Println(cfg.String())
+		return nil
+	} else {
+		return ioutil.WriteFile(sshConfigFile, []byte(cfg.String()), fileMode)
+	}
+}
+
+func removeHostEntries(hosts []*ssh_config.Host) (cleanedList []*ssh_config.Host) {
+	for i, host := range hosts {
+		if host.EOLComment != KubeVirtEOLComment {
+			cleanedList = append(cleanedList, hosts[i])
+		}
+	}
+	return cleanedList
+}
+
+func generateHostEntries(programName string, currentContext string, objects []unstructured.Unstructured) (hosts []*ssh_config.Host, err error) {
+	for _, obj := range objects {
+		host, err := generateHostEntry(programName, currentContext, &obj)
+		if err != nil {
+			return nil, err
+		}
+		hosts = append(hosts, host)
+	}
+	return hosts, nil
+}
+
+func generateHostEntry(programName string, currentContext string, obj *unstructured.Unstructured) (*ssh_config.Host, error) {
+	resource := ""
+	if obj.GetKind() == k6sv1.VirtualMachineInstanceGroupVersionKind.Kind {
+		resource = "vmi"
+	} else if obj.GetKind() == k6sv1.VirtualMachineGroupVersionKind.Kind {
+		resource = "vm"
+	} else {
+		return nil, fmt.Errorf("unsupported object kind: %s", obj.GetKind())
+	}
+	name := fmt.Sprintf("%s/%s.%s.%s", resource, obj.GetName(), obj.GetNamespace(), strings.ReplaceAll(currentContext, "@", "_"))
+	pattern, err := ssh_config.NewPattern(name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create host entry for %v: %v", name, err)
+	}
+	host := &ssh_config.Host{
+		EOLComment: KubeVirtEOLComment,
+		Patterns:   []*ssh_config.Pattern{pattern},
+		Nodes: []ssh_config.Node{
+			&ssh_config.KV{
+				Key:   "ProxyCommand",
+				Value: fmt.Sprintf("%s port-forward --context %s --stdio %s/%s.%s %s", programName, currentContext, resource, obj.GetName(), obj.GetNamespace(), "%p"),
+			},
+		},
+	}
+	return host, nil
+}
+
+func loadSSHConfig(path string) (*ssh_config.Config, os.FileMode, error) {
+	var fileMode os.FileMode = 0600
+	fileInfo, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return &ssh_config.Config{}, fileMode, nil
+	} else if err != nil {
+		return nil, fileMode, fmt.Errorf("failed to determine if %s exists: %v", path, err)
+	} else {
+		fileMode = fileInfo.Mode()
+	}
+
+	config, err := os.Open(path)
+	if err != nil {
+		return nil, fileMode, fmt.Errorf("could not open %s: %v", path, err)
+	}
+	defer config.Close()
+	cfg, err := ssh_config.Decode(config)
+	if err != nil {
+		return nil, fileMode, fmt.Errorf("failed to decote %s: %v", path, err)
+	}
+	return cfg, fileMode, nil
+}
+
+func examples() string {
+	return `  # Add ssh entries for all VMs and VMIs existing in the current namespace and context:
+  {{ProgramName}} config-ssh
```

Comment:
Maybe add an example specifying a different namespace as I think that's very common.
Sadly we don't support `--all-namespaces` right?

Reply:
Added a `--all-namespaces` option and added more examples.